include/radio.hpp

```diff
@@ -21,21 +21,31 @@ namespace bzzz
 
     private:
         /**
-         * Channel    Variable
-         * 1          yaw rate
-         * 2          pitch
-         * 3          throttle
-         * 4          roll
-         * 5          Switch C
-         * 6          Trimmer VRA
-         * 7          Trimmer VRC
-         * 8          Trimmer VRB
-         * 9          Switch B (for arming)
-         * 10         Switch A (kill)
-         * 11         Switch D
-         * 12         Trimmer VRE
-         */
-        int m_channelData[16];
+         * Array of reference data from radio with information as follows
+         * index    Variable
+         * 0          yaw rate in rad/s
+         * 1          pitch in rad
+         * 2          roll in rad
+         * 3          throttle in PWM range [1000, 2000]
+         * 4          % trim A
+         * 5          % trim B
+         * 6          % trim C
+         * 7          % trim E
+         */
+        float m_dummyRefData[8];
+        float m_refData[8];
+
+        /**
+         * The last part of the radio data from R-Pi is a 5-bit data encoding 
+         * the positions of switches as follows
+         * bit index    Variable
+         * 4(MSB)     Switch B (Arm switch  ){1 if switch_is_on else 0}
+         * 3          Switch A (Kill switch ){1 if switch_is_on else 0}
+         * 2, 1       Switch C (3-way switch){00 for position DOWN, 01 for position MID, 10 for position UP}
+         * 0(LSB)     Switch D (2-way switch){1 if switch_is_on else 0}
+        */
+        int m_dummyEncodedSwtchsData, m_encodedSwitchesData;
```

Comment:
One declaration per line:

```c++
int m_dummyEncodedSwtchsData;
int m_encodedSwitchesData;
```

and document each variable separately.

I'm not sure what you mean by "dummy"?

Reply:
We store the received data from R-Pi in these dummy variables. These dummy variables serve as a substitute (in place of using the actual `m_refData` array and `m_encodedSwitchesData` variable). We then check these dummy variables for corruption. If the data received is un-corrupted, we will copy them to the actual variables otherwise the previous data persists in the actual variables.

Reply:
I will make the suggested changes.

Reply:
Done.